### PR TITLE
💰 Order Approval | Basic Requisition Management Project

### PR DESCRIPTION
**Pull request by shauke**
**:moneybag: Order Approval | Basic Requisition Management**

The order approval functionality consists of the following parts:
- basics
- requisition, approval and budget management in the myAccount section
- extend the checkout to submit a basket for approval

basics:
- [x] `requisition-management` project
- [x] routing
- [x] breadcrumbs
- [x] requisition model (basic version)
- [x] state management / facade
- [x] service
- [x] permission checks
- [x] unit testing
- [x] configuration service toggle
- [x] approve not always working yet (not for others?)
- [x] breadcrumb for order detail translation key
- [x] requisition detail with context facade
- [x] configuration toggle (maybe extra task)

myAccount:
- [x] Approval/Requisitions List 
- [x] Approval/Requisition Details
- [x] Approve/Reject a requisition
- [x] Manage budgets for users

Checkout:
- [x] Display Approval required link and dialog in the checkout
- [x] Submit a requisition and display Thank you for your requisition page

*Limitations*
- Currently the cost center approval is not supported, i.e. you still cannot order with all of our demo users. Bio Tech users (ckitera, agoro, bassange, dhanssen) are suitable to demonstrate the functionality.
- Budgets can only be maintained in one currency

*Notes*
- direct routing/browser refresh support
- requisitions state as part of the customer state that would automatically take care of cleaning the store on logout?
- list REST call for listings, details call only for the selected details view

## PR Type

[x] Feature

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

### CODE REVIEW

File: projects/requisition-management/src/app/components/requisition-summary/requisition-summary.component.html
Comment (iwiederhold) [inline projects/requisition-management/src/app/components/requisition-summary/requisition-summary.component.html:67]: Approval Badges

Please use other badges for the approval status: 
<span class="badge badge-success">Success</span> for APPROVED
<span class="badge badge-danger">Danger</span> for REJECTED
<span class="badge badge-warning">Warning</span> for PENDING

**Review by iwiederhold — CHANGES_REQUESTED** [review]

**Review by M-Behr — CHANGES_REQUESTED** [review]
The basic functionality works really good. Thanks for the great teamwork! 👍 

I found some minor issues:
Requisition Details
Requisition Details:
- rename heading in Requisition Details
- remove pending status of last approver => it is needed, when cost center approval is implemented => then this "Approver(s): Conny Kitera (approved), Ralf Ulitzsch (pending)" is possible and then the user should know who has already approved and who has not

Order Details_Anpassungen
Order Details:
- some little adaptions: Order Number => Order No.; missing badge
- remove blank characters next to brackets

Order History vs Requisitions
Phone View:
- ":" is missing after labels
- differences between order list and requisition/approval lists => order total is displayed in first row/last row (My mistake! Sorry! I thought this did not fit into the first row.); also the spaces between lines are different => Can we unify this? I suggest: Order total in first line (like order history) and smaller spaces between lines (like requisitions and approvals)

ToDo's for later:
- add amount of rows (items) in other tables, e.g. order history, order templates, quoting, users.
- redesign budgets in buyer approval

**Comment by SGrueber** [comment]
To improve the responsive design of the order list is not part of this story. You can find a card for it in the technical debts project.

**Review by M-Behr — CHANGES_REQUESTED** [review]
The functionality works very well and the feature looks great! 👍 
I found only some naming convention issues and some inconsistencies:

**Checkout - Approval Details Dialog**
- missing sentence points when listing the conditions

**Approval / Requisition List**
- missing colons after label in phone view

**Approval / Requisition Details**
- "Date of Approval :" => "Date of Approval:" _(space character between word & colon)_
- if approved => order total is displayed in already spent bar, so no additional bar is needed in the diagram (this is also the reason why this "including this order" is not displayed in approved case)
- "Monthly budget:" => "Budget (monthly)" _(only for consistency reasons, see user management)_

**User Management - Popover for bar diagram**
- Labels in title case
- $ is missing for spent value
- "Monthly spend limit" => "Budget (monthly)"
- "spent" => "Already Spent" _(only for consistency reasons, see buyer approval)_

**User Management - User Details**
- "Budget (monthly) :" => "Budget (monthly):" _(space character between word & colon)_
- "No order spend limit defined." and "No budget defined." or use "Unlimited" by both _(currently it is a mix between these both possibilities)_

**User Management - Edit Budget**
- USD => $ _(only for consistency reasons)_

**Review by shauke — COMMENTED** [review]
- [x] requisition submission is not showing a loading animation for the whole process (the two combined POST REST calls), its only loading for the validation and after this it might take some time before the requisition is actually finished but one is not getting feedback of this ongoing process
image

**Review by M-Behr — CHANGES_REQUESTED** [review]
Some points are still open from the last review (see the checklist in my last comment).
There are some new issues on the User Management page:

- [x] The "Budget:" label is missing in case of unlimited. => add the label
- [x] The "Unlimited" value is displayed left. => move it to right
- [x] "Budget (weekly)::" => "Budget (weekly):" _(duplicate colons)_
User Management

**Review by iwiederhold — DISMISSED** [review]

**Review by M-Behr — CHANGES_REQUESTED** [review]
These points are still open:
 - [x] "Date of Approval :" => "Date of Approval:" _(remove space character between word & colon)_
 - [ ] "Budget (monthly) :" => "Budget (monthly):" _(remove space character between word & colon)_

Approval details

File: projects/requisition-management/src/app/store/requisitions/requisitions.reducer.ts
Comment (shauke) [inline projects/requisition-management/src/app/store/requisitions/requisitions.reducer.ts:50]: The `unsetLoadingAndErrorOn` function should be used.

**Review by shauke — COMMENTED** [review]
The overall functionality and implementation looks fine, only a few small things to improve. 👍

File: projects/requisition-management/src/app/exports/lazy-checkout-receipt-requisition/lazy-checkout-receipt-requisition.component.ts
Comment (shauke) [reply projects/requisition-management/src/app/exports/lazy-checkout-receipt-requisition/lazy-checkout-receipt-requisition.component.ts:65]: After talking to Max the `lazy-checkout-receipt-requisition` component should be removed.

File: src/app/core/store/customer/basket/basket.reducer.ts
Comment (SGrueber) [reply src/app/core/store/customer/basket/basket.reducer.ts:230]: this has nothing to do with this branch, only submittedBasket has been added

File: projects/requisition-management/src/index.html
Comment (SGrueber) [reply projects/requisition-management/src/index.html:5]: as long as we cannot use the requisition management as standalone project is doesn't matter

**Review by M-Behr — DISMISSED** [review]
Good job! 👍

**Pull request merged**
